src/Symfony/Component/HttpFoundation/Response.php

```diff
@@ -888,22 +897,27 @@ public function getLastModified()
      *
      * Passing null as value will remove the header.
      *
-     * @param \DateTime|null $date A \DateTime instance or null to remove the header
+     * @param \DateTimeInterface|null $date A \DateTime instance or null to remove the header
      *
      * @return $this
      *
      * @final since version 3.2
      */
-    public function setLastModified(\DateTime $date = null)
+    public function setLastModified(\DateTimeInterface $date = null)
     {
         if (null === $date) {
             $this->headers->remove('Last-Modified');
-        } else {
-            $date = clone $date;
-            $date->setTimezone(new \DateTimeZone('UTC'));
-            $this->headers->set('Last-Modified', $date->format('D, d M Y H:i:s').' GMT');
+
+            return $this;
         }
 
+        if ($date instanceof \DateTime) {
+            $date = \DateTimeImmutable::createFromMutable($date);
+        }
+
+        $date = $date->setTimezone(new \DateTimeZone('UTC'));
```

Comment:
Correct me if I'm wrong, but `DateTimeInterface` doesn't have a `setTimezone` method from what I can tell.

Reply:
It doesn't. That's why the block above that line makes sure we're dealing with a `DateTimeImmutable` here.

Reply:
To explain this a bit more: The sole purpose of `DateTimeInterface` is to allow type-hinting for objects that are either `DateTime` or `DateTimeImmutable`. Those are the only two implementations of that interface and the php runtime doesn't allow userland code to directly implement the interface.

So at the beginning of the method, `$date` has to be either, `null`, `DateTime` or `DateTimeImmutable`. For `null`, we have an early exit. A `DateTime` instance is converted to `DateTimeImmutable`.

So right here, `$date` has to be an instance of `DateTimeImmutable` which contains the method `setTimezone()`.

Reply:
`the php runtime doesn't allow userland code to directly implement the interface.`

Does this include extensions? Regardless of php allowing it or not, it looks weird like this. Should probably throw an exception or ignore the setter when it's not supported.

Reply:
We don't need to handle something that doesn't happen.  that's just adding visual noise. to me it's good as it is

Reply:
> Does this include extensions?

An extension could probably do that, but it would be a very odd thing to do.

I could not even write a test for the case that we have a `DateTimeInterface` instance that is neither a `DateTime` nor a `DateTimeImmutable` instance. I don't think, we need to handle it.